### PR TITLE
allow user input on credit cards

### PR DESCRIPTION
**Pull request by oriarditi**

### CODE REVIEW

File: lib/src/main/java/tk/zielony/randomdata/RandomData.java [hunk 1]
```diff
@@ -31,25 +31,48 @@ public int compareTo(@NonNull GeneratorWithPriority generatorWithPriority) {
...
     private List<GeneratorWithPriority> generators = new ArrayList<>();
+    private String userInput = null;
 
     public void addGenerators(Generator[] generators) {
         for (Generator g : generators)
             this.generators.add(new GeneratorWithPriority(g, Priority.Normal));
     }
 
+    public void addGenerators(Generator[] generators, String userInput) {
+        this.userInput = userInput;
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/RandomData.java]: Should call here addGenerators(generators); instead of repeating the function.

File: lib/src/main/java/tk/zielony/randomdata/RandomData.java [hunk 2]
```diff
@@ -31,25 +31,48 @@ public int compareTo(@NonNull GeneratorWithPriority generatorWithPriority) {
...
+        for (Generator g : generators)
+            this.generators.add(new GeneratorWithPriority(g, priority));
+    }
+
     public void addGenerator(Generator generator) {
         generators.add(new GeneratorWithPriority(generator, Priority.Normal));
     }
 
+    public void addGenerator(Generator generator, String userInput) {
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/RandomData.java]: The method is redundant, should call addGenerator(generator, userInput, priority);

File: lib/src/main/java/tk/zielony/randomdata/RandomData.java [hunk 3]
```diff
@@ -31,25 +31,48 @@ public int compareTo(@NonNull GeneratorWithPriority generatorWithPriority) {
...
+    }
+
     public void addGenerators(Generator[] generators, Priority priority) {
         for (Generator g : generators)
             this.generators.add(new GeneratorWithPriority(g, priority));
     }
 
+    public void addGenerators(Generator[] generators, String userInput, Priority priority) {
+        this.userInput = userInput;
+        for (Generator g : generators)
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/RandomData.java]: Check whether the method from line 41 is required.
The only difference between both functions is that the first one sets the priority to Normal by default
addGenerators(Generator[] generators, String userInput, Priority priority)

File: lib/src/main/java/tk/zielony/randomdata/RandomData.java [hunk 4]
```diff
@@ -31,25 +31,48 @@ public int compareTo(@NonNull GeneratorWithPriority generatorWithPriority) {
...
+            this.generators.add(new GeneratorWithPriority(g, priority));
+    }
+
     public void addGenerator(Generator generator) {
         generators.add(new GeneratorWithPriority(generator, Priority.Normal));
     }
 
+    public void addGenerator(Generator generator, String userInput) {
+        this.userInput = userInput;
+        generators.add(new GeneratorWithPriority(generator, Priority.Normal));
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/RandomData.java]: Call addGenerator(Generator generator, Priority priority);

File: lib/src/main/java/tk/zielony/randomdata/RandomData.java [hunk 5]
```diff
@@ -31,25 +31,48 @@ public int compareTo(@NonNull GeneratorWithPriority generatorWithPriority) {
...
+    }
+
     public void addGenerator(Generator generator, Priority priority) {
         generators.add(new GeneratorWithPriority(generator, priority));
     }
 
+    public void addGenerator(Generator generator, String userInput, Priority priority) {
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/RandomData.java]: Check whether the method from line 62 is required.
The only difference between both functions is that the first one sets the priority to Normal by default

File: lib/src/main/java/tk/zielony/randomdata/person/StringEmailGenerator.java [hunk 6]
```diff
@@ -25,10 +25,10 @@ protected Matcher getDefaultMatcher() {
     }
 
     @Override
-    public String next(DataContext context) {
+    public String next(DataContext context, String userInput) {
         String name = context.getName();
         if (name == null) {
-            name = nameGenerator.next(context);
+            name = nameGenerator.next(context, null);
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/person/StringEmailGenerator.java]: I think we should pass the recursive the userInput
name = nameGenerator.next(context, userInput);

File: lib/src/main/java/tk/zielony/randomdata/finance/StringCardNumberGeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class StringCardNumberGeneratorTest {
+
+    StringCardNumberGenerator generator = new StringCardNumberGenerator();
+
+    @Test
+    public void userInput() throws Exception {
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/finance/StringCardNumberGeneratorTest.java]: Change the test name to reflect the test's purpose (something like "TestCardNumberContainsUserInput")

File: lib/src/main/java/tk/zielony/randomdata/finance/StringCardNumberGeneratorTest.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class StringCardNumberGeneratorTest {
+
+    StringCardNumberGenerator generator = new StringCardNumberGenerator();
+
```
Comment (tonyfelik) [inline lib/src/main/java/tk/zielony/randomdata/finance/StringCardNumberGeneratorTest.java]: Add test where the userInput is null
